### PR TITLE
Change Logic for Handling getMyData API Call

### PR DESCRIPTION
**Pull request by anthonypetersen**
Addresses the following Issue:

https://github.com/episphere/connect/issues/638

Tied to PR:

https://github.com/episphere/connectFaas/pull/365

Technical Changes

* added function `hasUserData()` to check for `200` code and a non-empty data object
* updated all instances of analyzing return object of calls to `getMyData()` with a call to `hasUserData()`

### CODE REVIEW

File: js/shared.js [hunk 1]
```diff
@@ -139,14 +139,19 @@ export const getMyData = async () => {
...
+            Authorization: 'Bearer ' + idToken,
+        },
     });
 
     return response.json();
 };
 
+export const hasUserData = (response) => {
+
+    return response.code === 200 && Object.keys(response.data) > 0;
```
Comment (JoeArmani) [inline js/shared.js:152]: Is this the intended comparison? I think we may be looking for the `.length` operator:
```javascript
return response.code === 200 && Object.keys(response.data).length > 0;
```

**Review by JoeArmani — APPROVED** [review]
Added hasUserData check to
•userProfile() in app.js (sub for code === 200 check)
•renderUserProfile() in form.js (remove siteId check)
•verifyUserDetails() in event.js (sub for code === 200 check)
•renderAgreements() in agreements.js (sub for code === 200 check)
•consentWelcomePage(), consentConsentPage(), consentFinishedPage(), and consentSubmit() in consent.js (sub for code ===200 check or add in missing check)
•whereAmIInDashboard() in homePage.js (sub for code === 200 check)
•renderPaymentPage() in payments.js (sub for .data check)
•questionnaire() in questionnaire.js (sub for code === 200 check)
•renderSamplesPage() in samples.js (no check existed)
•renderSettingsPage() and refreshUserDataAfterEdit() in settings.js (sub for code === 200 check)
•renderSupportPage() in support.js (sub for .data check)
•renderVerifiedPage() in verifiedPage.js (sub for .data check)

Added hasUserData() function in shared.js

My only question involves using the .length operator in hasUserData()

File: js/shared.js (hunk 1 above)
Comment (anthonypetersen) [reply js/shared.js:152]: Great catch!

**Comment by jhflorey** [comment]
It seems there are some changes that are beyond the scope of this PR, such as
- Removing `errorMessageNumbers` from `js/event.js`
- Removing  `hideAnimation`, `showAnimation` from  `js/pages/consent.js`
=> IMHO, we should have another PR to remove all used variables as well as refactor code.

To summarize, except for  the problem of checking the length that Joe caught. This PR mainly replaces ` myData.code === 200` with `hasUserData` which will check for 200 code and a non-empty data object. It looks good for me

**Review by jhflorey — APPROVED** [review]

**Review by we-ai — APPROVED** [review]
Looks good.

**Pull request merged**